osf/models/institution.py

```diff
@@ -8,19 +8,43 @@
 
 
 class Institution(Loggable, base.ObjectIDMixin, base.BaseModel):
+
     # TODO Remove null=True for things that shouldn't be nullable
-    banner_name = models.CharField(max_length=255, null=True, blank=True)
+    # e.g. CharFields should never be null=True
+
+    name = models.CharField(max_length=255)
+    description = models.TextField(blank=True, null=True)
+
+    # TODO Could `banner_name` and `logo_name` be a FilePathField?
+    banner_name = models.CharField(max_length=255, blank=True, null=True)
+    logo_name = models.CharField(max_length=255, blank=True, null=True)
+
+    # The protocol which is used to delegate authentication.
+    # Currently, we have `CAS`, `SAML`, `OAuth` available.
+    # For `SAML`, we use Shibboleth.
+    # For `CAS` and `OAuth`, we use pac4j.
+    # Only institutions with a valid delegation protocol show up on the institution login page.
+    DELEGATION_PROTOCOL_CHOICES = (
+        ('cas-pac4j', 'CAS by pac4j'),
+        ('oauth-pac4j', 'OAuth by pac4j'),
+        ('saml-shib', 'SAML by Shibboleth'),
+        ('', 'No Delegation Protocol'),
+    )
+    delegation_protocol = models.CharField(max_length=15, choices=DELEGATION_PROTOCOL_CHOICES, blank=True)
```

Comment:
Add `delegation_protocol` using `Field.choices`, which is used by CAS to determine how to delegate the authentication.

Other changes of this file is just moving codes around and update comments.